### ISSUE TITLE
Human friendly IDs + next steps

### ISSUE THREAD

**Issue by ieguinoa** [refgenie/plantref#11]
Following on your comment about using common/short names, aliases, or adapt the names to be more human friendly, I think we can come up with a list like this, at leas for some of the species 
My question is how this would be implemented without loosing the information that is currently in the name, were you thinking of creating a kind of alias for the name? or by creating a tag to the fasta asset?
If using tags then we can also use it to group versions of a genome, leaving the "core" part of the id as genome name and move the build id as tag.
An example could be Zea mays. This could be the genome name (Zea_mays_B73, or simply ZeaMays as B73 is the main reference line) but it has a few entries in the genomes list, this info should go in the tags. eg the tags list for the fasta asset would be: B73_AGP_v4_0 and B73_5b_60
Normally all these would be different versions, we will make sure that these groupings make sense in terms of versions and that we can identify the latest one which will be referred by default also. In case of branches of the genome, like different strains or such there would be a different genome name. e.g , ZeaMays PH207 is a different line as B73 so it would go into a genome named PH207 and a single fasta asset with tag v1_0.
Thus, when adding a new version of a genome (e.g we are adding ZeaMays v5.0), the upgrading process would mean adding this asset tag 5.0 and also change the default to it. This way, when someone runs `refgenie pull ZeaMays:fasta` they would get the default->5.0. 
From the point of view of the data curation we need to make sure that all default assets for a genome are compatible.

Also a related question about next assets to include: what is the best way to have linked assets that do not have a natural parent/children relationship?
If we start including annotations we may have cases where multiple annotations are associated with the same genome fasta and version. e.g we can have zea_mays_GTF_annotation_v1.0 and zea_mays_GTF_annotation_v2.0
I think a possible way to do this is to use the genome fasta asset as a (optional) parent in the GTF annotation build recipe.  
It is not strictly used during the build, but can be used to check compatibility and to keep track of the association. 
Or would it be possible to add an option to set "associated-assets" and then link an annotation with a genome fasta in a specific tag?

**Comment by nsheff** [comment]
> My question is how this would be implemented without loosing the information that is currently in the name, were you thinking of creating a kind of alias for the name? or by creating a tag to the fasta asset?

You should use the short human readable identifier for the alias, and leave the more detailed information in the description. If you look at the [server instance](http://plantref.databio.org) you can see that right now they're basically just duplicated. 
genome descriptions show up here: https://github.com/refgenie/plantref/blob/master/asset_pep/genome_descriptions.csv

Tags shouldn't be used in this way; they are for versions. But for a FASTA file, another version should really be considered another genome. In the past, we debated with the possibility of eliminating tags for fasta files. For example, we use "hg38" and "hg19" to refer to different version of the human assembly. Those aren't just different tags, they're different aliases and different genomes.

>An example could be Zea mays.... 

I'd suggest you have something like `ZeaMaysAGP` and `ZeaMays5b60` and `ZeaMaysPH207` as your human-readable aliases. Those are separate genomes, separate fasta files, and so they don't just get tags, they get their own genome alias.

> From the point of view of the data curation we need to make sure that all default assets for a genome are compatible.

Exactly -- tags wouldn't do this well... you need a new bowtie2/bwa index and gtf annotation file for each of these fasta files, so they're separate genome entries, not just tags.

> what is the best way to have linked assets that do not have a natural parent/children relationship?
> If we start including annotations we may have cases where multiple annotations are associated with the same genome fasta and version. e.g we can have zea_mays_GTF_annotation_v1.0 and zea_mays_GTF_annotation_v2.0

This is where tags are useful. Because they are both compatible with the same `ZeaMays` genome, but are just different versions of the same asset. So here you'd have `ZeaMays/gtf:v1` and `ZeaMays/gtf:v2`.

> I think a possible way to do this is to use the genome fasta asset as a (optional) parent in the GTF annotation build recipe.
> It is not strictly used during the build, but can be used to check compatibility and to keep track of the association.
> Or would it be possible to add an option to set "associated-assets" and then link an annotation with a genome fasta in a specific tag?

There are two different concepts here. One is the parent-child relationship between assets, the other is the namespace relationship. What you're describing here should just be a namespace relationship, not a parent-child relationship.

A parent-child relationship indicates that an asset was used to build another asset. for example, a this drosophila bowtie2 index asset was built from a fasta asset: 

![image](https://user-images.githubusercontent.com/5271009/99832967-5be31c80-2b2f-11eb-977f-a6caf25ee42d.png)

But this drosophila gtf file was *not* produced *from* the fasta asset:
![image](https://user-images.githubusercontent.com/5271009/99832993-6e5d5600-2b2f-11eb-809a-c7f615afdbba.png)

Yet, it's compatible. it still lives in the `dm6` namespace.

So, the answer to your question is: yes, it is possible to include linked asses with natural parent-child relationships -- this is exactly what the genome namespaces accomplish. GTF assets don't have "optional parents" to indicate the association to the fasta file; this is a looser association that's specified by belonging to the same genome namespace. parent-child strictly means an asset was an input for building another asset.

**Comment by ieguinoa** [comment]
thanks for the detailed reply, I think it's clear now. 
I do have a couple more short questions before formatting the annotations and transcriptome input files: 
I see that in the refgenomes builds (https://github.com/refgenie/refgenomes.databio.org/blob/master/asset_pep/recipe_inputs.csv) the transcriptomes have 2 different kind of entries, at least for some genomes: there is a different namespace (e.g hg18_cdna) and there is also the same file created as an asset (fasta_txome) under the main namespace hg18, is there a reason for this duplication? given that the asset recipe for both fasta and fasta_txome assets is the same and in this case (hg18) the input file URL is also the same. Which method should I use? 

And about the annotations, as we discussed before, there would be genomes with  more than 1 annotation file and this can be handled with tags, how can I specify tags in the CI files? is it parsed from the sample name in recipe_inputs.csv if I write something like  ZeaMaysV4-gencode_gtf:v1 ?

**Comment by ieguinoa** [comment]
ping @nsheff 
sorry I replied here and not in the email, just remembered you may not get the notification

**Comment by nsheff** [comment]
Hi @ieguinoa got it -- I've been thinking about it. I have a solution for the tag versions, I'll just need to add the tag in to the csv file and update the interface file, shouldn't take too much. I'll get back to you shortly after I test it out

I am still thinking of the best answer the first question. This will become easier once we have finished https://github.com/refgenie/refgenie/issues/198 but we are not quite there yet.

**Comment by nsheff** [comment]
Alright, I've updated the repo to accommodate tags. Just make sure to:

1. put the correct entry in the `tag` column in `assets.csv`
2. in the `recipe_inputs.csv`, make sure to name the `sample_name` as `{genome}-{asset}-{tag}` -- this will allow you to specify different input files for different versions (tags) of the same asset. 

That's it.

On the duplicate fasta_txome both as {genome}/fasta_txome *and* {genome}_cdna/fasta -- for right now. Right now we're using both. We need the _cdna one to build indexes on just the txome, but we need it under the primary {genome} namespace to build indexes like salmon indexes that require both. So, for right now, I think you can either:

1. do both, if you want it to be able to used in both scenarios
2. if you don't need the new salmon indexes that use both genome and transcriptome, just use the _cdna approach.

This change once refgenie/refgenie#198 is solved, and there will be a better way to do this, but this will be awhile so I think we should move forward with the current system. It will be easy to adapt systematically when we get to that point.

anything else?

**Comment by nsheff** [comment]
Hi @ieguinoa -- wanted to revisit this as we just released a new version of refgenie server with a new approach to genome identifiers. Have you had a chance to think about this yet? Thanks

**Comment by ieguinoa** [comment]
Hi @nsheff , I'm back at this. Just made a PR with a first manual intervention of the genome names, definitely helps with simplicity having a shorter, simplified name with details in the description only.
I've also put a few of the genomes in stage for the moment, are a bit too old and may not be worth adding them, at least for the moment.
Lets see if this PR passes the CI and then I will make the link with annotation and transcriptome, that should be straightforward using tags once the genome namespaces are settled.

**Comment by nsheff** [comment]
alright great. We actually don't have CI set up for this yet, it's just me checking at this point. But yeah, eventually this would all be deploying automatically...

**Comment by nsheff** [comment]
Ok I think that's great. Are you going to add some tagged assets?

**Comment by ieguinoa** [comment]
yes, working on that. annotations and cdna definitely need to use tags. since these may require a bit of manual work to get the association with the namespace correctly (at least for the annotation) I may add a few cases first to make sure it all works fine and then add the rest.

**Comment by nsheff** [comment]
ok that's a good idea. if you can add a few of them, then I'll rebuild everything and launch the server and you can check. if it works, then add the rest

**Comment by ieguinoa** [comment]
I'm adding a few fasta_txome assets, still all default. 
I will be adding annotations assets later. About this I was wondering what is the right asset type/recipe for a common gff3 type annotation. So far I've been using gencode_gtf as the recipe just copies the input file but dont know if there is now any other more appropriate asset type.

**Comment by nsheff** [comment]
On the gffs, there have been no updates. Let's just use the `gencode_gtf` for this small test. Like the fasta_txome assets above, this will change when we revamp the recipe system, but it will be useful to have a few items in here to test with. So, go ahead and add those when you're ready.
